Objective-C/CBLDatabase.h

```diff
@@ -417,6 +417,18 @@ typedef NS_ENUM(uint32_t, CBLMaintenanceType) {
  */
 - (BOOL) createIndex: (CBLIndex*)index withName: (NSString*)name error: (NSError**)error;
 
+/**
+ Creates an index using IndexConfiguration, which could be a value index or a full-text search index with the given name.
+ Creating a new different index with an existing index name will replace the old index;
+ creating the same index with the same name will be no-ops.
+ 
+ @param config The index configuration
+ @param name The index name.
+ @param error error On return, the error if any.
+ @return True on success, false on failure.
+ */
+- (BOOL) createIndexWithConfig: (CBLIndexConfiguration*)config
+                      withName: (NSString*)name error: (NSError**)error;
```

Comment:
maybe just 'name' instead of 'withName' to be consistent with others. We always have one 'with'.